x/marketmap/types/ticker.go

```diff
@@ -0,0 +1,94 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	// DefaultMaxDecimals is the maximum number of decimals allowed for a ticker.
+	DefaultMaxDecimals = 36
+	// DefaultMinProviderCount is the minimum number of providers required for a
+	// ticker to be considered valid.
+	DefaultMinProviderCount = 1
+)
+
+// NewTicker returns a new Ticker instance. A Ticker represents a price feed for
+// a given asset pair i.e. BTC/USD. The price feed is scaled to a number of decimal
+// places and has a minimum number of providers required to consider the ticker valid.
+func NewTicker(base, quote string, decimals, minProviderCount uint64) (Ticker, error) {
+	t := Ticker{
+		Base:             base,
+		Quote:            quote,
+		Decimals:         decimals,
+		MinProviderCount: minProviderCount,
+	}
+
+	if err := t.ValidateBasic(); err != nil {
+		return Ticker{}, err
+	}
+
+	return t, nil
+}
+
+// String returns a string representation of the Ticker.
+func (t *Ticker) String() string {
+	return fmt.Sprintf("%s/%s", t.Base, t.Quote)
+}
```

Comment:
One thing to note is that since this is the same exact thing as `oracletypes.CurrencyPair`, we should probably only use one or the other.

Reply:
imo, we make a shared types package and put this in it.

`./pkg/types/ticker.pb.go`
`./pkg/types/ticker.go`

then we use `slinkytypes.Ticker` throughout

Reply:
hmm

Reply:
thats not a bad, im fine changing the Ticker to be Currency Pair since we're going to have parity anyways